src/app/services/tree-node.service.ts

```diff
@@ -185,17 +283,27 @@ export class TreeNodeService {
    * @returns {Concept}
    */
   public getConceptFromTreeNode(treeNode: TreeNode): Concept {
-    let concept = new Concept();
-    const tail = '\\' + treeNode['name'] + '\\';
-    const fullName = treeNode['fullName'];
-    let head = fullName.substring(0, fullName.length - tail.length);
-    concept.label = treeNode['name'] + ' (' + head + ')';
-    concept.path = treeNode['conceptPath'];
-    concept.type = <ConceptType> treeNode['type'];
-    concept.code = treeNode['conceptCode'];
-    concept.fullName = treeNode['fullName'];
-    concept.name = treeNode['name'];
-    return concept;
+    if (treeNode['name'] &&
```

Comment:
You can use `treeNode.name` instead.

Reply:
yes, I am actually thinking of refactoring the tree node type, it is something that primeNg provides, which turns out to be more and more ill-fitted for glowing bear, but not priority now